src/BalanceLaws/interface.jl

```diff
@@ -316,18 +262,11 @@ Compute reverse indefinite integral along stack.
 function reverse_indefinite_stack_integral! end
 
 # Internal methods
-number_state_conservative(m::BalanceLaw, FT) =
-    varsize(vars_state_conservative(m, FT))
-number_state_auxiliary(m::BalanceLaw, FT) = varsize(vars_state_auxiliary(m, FT))
-number_state_gradient(m::BalanceLaw, FT) = varsize(vars_state_gradient(m, FT))
-number_state_gradient_flux(m::BalanceLaw, FT) =
-    varsize(vars_state_gradient_flux(m, FT))
-num_gradient_laplacian(m::BalanceLaw, FT) =
-    varsize(vars_gradient_laplacian(m, FT))
-num_hyperdiffusive(m::BalanceLaw, FT) = varsize(vars_hyperdiffusive(m, FT))
-num_integrals(m::BalanceLaw, FT) = varsize(vars_integrals(m, FT))
-num_reverse_integrals(m::BalanceLaw, FT) =
-    varsize(vars_reverse_integrals(m, FT))
+number_states(m::BalanceLaw, st::AbstractStateType, FT) =
+    varsize(vars_state(m, st, FT))
```

Comment:
```suggestion
number_states(m::BalanceLaw, st::AbstractStateType, FT=Int) =
    varsize(vars_state(m, st, FT))
```
The type doesn't matter here. So this will allow `number_states(m, st)` to work.